src/test/java/io/stargate/sgv2/jsonapi/api/v1/IndexingConfigIntegrationTest.java

```diff
@@ -0,0 +1,899 @@
+
+  private static final String denyOneIndexingCollection = "deny_one_indexing_collection";
+
+  private static final String denyManyIndexingCollection = "deny_many_indexing_collection";
+
+  private static final String denyAllIndexingCollection = "deny_all_indexing_collection";
+
+  private static final String allowOneIndexingCollection = "allow_one_indexing_collection";
+
+  private static final String allowManyIndexingCollection = "allow_many_indexing_collection";
+
+  @Nested
+  @Order(1)
+  class CreateCollectionAndData {
+    String insertData =
+        """
+              {
+                "insertOne": {
+                  "document": {
+                    "_id": "1",
+                    "name": "aaron",
+                    "$vector": [0.25, 0.25, 0.25, 0.25, 0.25],
+                    "address": {
+                      "street": "1 banana street",
+                      "city": "monkey town"
+                    }
+                  }
+                }
+              }
+                 """;
+
+    @Test
+    public void createCollectionAndData() {
+      String createDenyOneIndexingCollection =
+          """
+              {
+                "createCollection": {
+                  "name": "%s",
+                  "options" : {
+                    "vector" : {
+                      "size" : 5,
+                      "function" : "cosine"
+                    },
+                    "indexing" : {
+                      "deny" : ["address.city"]
+                    }
+                  }
+                }
+              }
+                  """
+              .formatted(denyOneIndexingCollection);
+      // create collection
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(createDenyOneIndexingCollection)
+          .when()
+          .post(NamespaceResource.BASE_PATH, namespaceName)
+          .then()
+          .statusCode(200)
+          .body("status.ok", is(1));
+      // insert data
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(insertData)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, denyOneIndexingCollection)
+          .then()
+          .statusCode(200);
+
+      String createDenyManyIndexingCollection =
+          """
+              {
+                "createCollection": {
+                  "name": "%s",
+                  "options" : {
+                    "vector" : {
+                      "size" : 5,
+                      "function" : "cosine"
+                    },
+                    "indexing" : {
+                      "deny" : ["name", "address"]
+                    }
+                  }
+                }
+              }
+                  """
+              .formatted(denyManyIndexingCollection);
+      // create collection
```

Comment:
Since you create multiple collections, insert multiple entries, may make sense to extract helper methods.
`AbstractCollectionIntegrationTestBase` actually has `createCollection()` and `insertDoc()` if you could that base class instead -- we could maybe refactor things a bit and move `createCollection(String namespace, String collectionName)` into that class, for better reuse, reduced code duplication.